source/op/cuda/CMakeLists.txt

```diff
@@ -14,14 +14,77 @@ SET(CMAKE_CXX_STANDARD 11)
+                        -gencode arch=compute_50,code=sm_50; # Quadro 620 1200
+                        -gencode arch=compute_52,code=sm_52; # Tesla M40 M40, Quadro M6000 M5000 M4000 M2000, TITAN X, GTX 980Ti 980 970 960 950
+                        -gencode arch=compute_53,code=sm_53; # Jetson TX1, Tegra X1
+                        -gencode arch=compute_60,code=sm_60; # Pascal – GP100/Tesla P100 – DGX-1 (Generic Pascal)
+                        -gencode arch=compute_61,code=sm_61; # Pascal - GTX 1080, GTX 1070, GTX 1060, GTX 1050, GTX 1030, Titan Xp, Tesla P40, Tesla P4, Discrete GPU on the NVIDIA Drive PX2
+                        -gencode arch=compute_70,code=sm_70; # Volta  - GV100/Tesla V100, GTX 1180 (GV104)
+                        -gencode arch=compute_75,code=sm_75; # Turing - RTX 2080, Titan RTX, Quadro R8000
+                        -O3; -Xcompiler -fPIC;
+        )
+elseif (${CUDA_VERSION_MAJOR} STREQUAL "9")
+    message(STATUS "CUDA major version is " ${CUDA_VERSION_MAJOR})
+    # nvcc flags
+    set(CUDA_NVCC_FLAGS -gencode arch=compute_30,code=sm_30;
+                        -gencode arch=compute_35,code=sm_35;
+                        -gencode arch=compute_37,code=sm_37;
+                        -gencode arch=compute_50,code=sm_50;
+                        -gencode arch=compute_52,code=sm_52; # Tesla M40, Tesla M40, Quadro M6000...
+                        -gencode arch=compute_53,code=sm_53; 
+                        -gencode arch=compute_60,code=sm_60; # Pascal – GP100/Tesla P100 – DGX-1 (Generic Pascal)
+                        -gencode arch=compute_61,code=sm_61; # Pascal - GTX 1080, GTX 1070, GTX 1060, GTX 1050, GTX 1030, Titan Xp, Tesla P40, Tesla P4, Discrete GPU on the NVIDIA Drive PX2
+                        -gencode arch=compute_70,code=sm_70; # Volta  - GV100/Tesla V100, GTX 1180 (GV104)
+                        -O3; -Xcompiler -fPIC;
+        )
+elseif (${CUDA_VERSION_MAJOR} STREQUAL "8")
+    message(STATUS "CUDA major version is " ${CUDA_VERSION_MAJOR})
+    # nvcc flags
+    set(CUDA_NVCC_FLAGS -gencode arch=compute_30,code=sm_30;
+                        -gencode arch=compute_35,code=sm_35;
+                        -gencode arch=compute_37,code=sm_37;
+                        -gencode arch=compute_50,code=sm_50;
+                        -gencode arch=compute_52,code=sm_52; # Tesla M40, Tesla M40, Quadro M6000...
+                        -gencode arch=compute_53,code=sm_53; 
+                        -gencode arch=compute_60,code=sm_60; # Pascal – GP100/Tesla P100 – DGX-1 (Generic Pascal)
+                        -gencode arch=compute_61,code=sm_61; # Pascal - GTX 1080, GTX 1070, GTX 1060, GTX 1050, GTX 1030, Titan Xp, Tesla P40, Tesla P4, Discrete GPU on the NVIDIA Drive PX2
+                        -O3; -Xcompiler -fPIC;
+        )
+elseif (${CUDA_VERSION_MAJOR} STREQUAL "7")
+    message(STATUS "CUDA major version is " ${CUDA_VERSION_MAJOR})
```

Comment:
It would be better to move this line out of the conditions, because it is the same for all branches.

Reply:
👌